ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/HiveCalciteUtil.java

```diff
@@ -1062,6 +1060,25 @@ public RexNode apply(RelDataTypeField input) {
     return HiveProject.create(input, copyInputRefs, null);
   }
 
+  public static boolean isConstant(RexNode expr) {
+    if (expr instanceof RexCall) {
+      RexCall call = (RexCall) expr;
+      if (call.getOperator() == SqlStdOperatorTable.ROW ||
+          call.getOperator() == SqlStdOperatorTable.ARRAY_VALUE_CONSTRUCTOR ||
+          call.getOperator() == SqlStdOperatorTable.MAP_VALUE_CONSTRUCTOR) {
```

Comment:
I think support to recognize `SqlMultisetValueConstructor` would need to be added in other places too in any case... We can leave it more explicit for the time being and generalize later on if needed. Fwiw this call is also very specific to Hive and how it handles the initialization of UDFs with parameters that are constant; otherwise, we would not need it.